diff --git a/mailman_groups.module b/mailman_groups.module
index 5e08ace..0d97b14 100644
--- a/mailman_groups.module
+++ b/mailman_groups.module
@@ -138,6 +138,9 @@ function _mailman_groups_update($lid, $gid, $hide = 1, $sync = 0) {
  * Delete an entry from the mailman_groups table.
  */
 function _mailman_groups_unlink($lid, $gid) {
+  // Tranform these strings into integers to go into the database
+  $lid = intval($lid);
+  $gid = intval($gid);
   $query = 'DELETE FROM {mailman_groups} WHERE lid = %d AND gid = %d';
   db_query($query, $lid, $gid);
 }
@@ -315,7 +318,11 @@ function _mailman_groups_alter_og_confirm_subscribe(&$form, $form_state) {
         if (!_mailman_groups_checkfunc('_mailman_manager_subscribe')) {
           return;
         }
-        _mailman_manager_subscribe($user, $lid, $user->mail, 'nodigest', '');
+        // First create the subscription..this is necessary as there is nothing doing this already.
+        // This function creates a subscription if one doesn't already exist and sets its status to 0.
+        _mailman_manager_get_subscriptions($account->uid, $lid);
+        // Then update the new or existing subscription to '3' - One email, not digest.
+        _mailman_manager_update_subscriptions($account->uid, $lid, '3', '0', $account->mail, '');
       }
     }
   }
